### PR TITLE
Support selecting non-word entry as a query

### PR DESCRIPTION
**Pull request by HyunggyuJang**
Currently, it assumes only words can be queried via dictionary. However, it isn’t necessarily true: Take the example of “TD;LR”, which is an proper entry in osx dictionary, but not a word. 

It can be relieved by allowing user to type a query that is not provided by completion.

### CODE REVIEW

**Comment by HyunggyuJang** [comment]
Org roam’s find file function would be a good source as it already has this feature.

**Comment by HyunggyuJang** [comment]
Here is the relevant part from org-roam: In [org-roam-find-file](https://github.com/org-roam/org-roam/blob/9065f6a999b98d4b495e3d8fa1fa4424eddd25a8/org-roam.el#L1576)

``` emacs-lisp
(org-roam-completion--completing-read "File: " completions
                                      :initial-input initial-prompt)
```

**Comment by HyunggyuJang** [comment]
Wasn't fault of completing function. 

See the following error message:
> anki-vocabulary--parse-english-dictionary: Args out of range: "TLDR: the service was horrible and the food took too long.

Parser was the culprit. Need to inspect carefully before jumping into the code.

**Comment by HyunggyuJang** [comment]
Need to improve parser from delimiter based to syntax aware parser.

See the problematic query result:

> TL;DR (also TLDR) abbreviation informal used as a dismissive response to a lengthy online post, or to introduce a summary of a lengthy post: TL;DR—I just scrolled all the way down but that took forever | TLDR: the service was horrible and the food took too long.
> ORIGIN early 21st century: abbreviation of too long, didn't read or too long, don't read.
